src/server/req_delete.c

```diff
@@ -703,12 +814,8 @@ req_deletejob(struct batch_request *preq)
 						log_eventf(PBSEVENT_DEBUG, PBS_EVENTCLASS_JOB, LOG_INFO, pjob->ji_qs.ji_jobid, msg_job_history_delete, preq->rq_user, preq->rq_host);
 						job_purge(pjob);
 					} else {
-						if (dup_br_for_subjob(preq, pjob, req_deletejob2) == 2) {
-							preq->rq_ind.rq_deletejoblist.jobid_to_resume = j;
-							preq->rq_ind.rq_deletejoblist.subjobid_to_resume = i;
-							--preq->rq_refct;
-							return;
-						}
+						if (dup_br_for_subjob(preq, pjob, req_deletejob2) == 2)
+							continue;
```

Comment:
same comment here, don't we need to return/defer the request instead of continue?

Reply:
No delete list can continue. The subjob has it's own request structure which will work on its own.